### ISSUE TITLE
[native] AbstractTestNativeGeneralQueries.testDecimalArithmetic and similar are failing

### ISSUE THREAD

**Issue by mbasmanova** [prestodb/presto#19569]
CC: @karteekmurthys @aditi-pandit @majetideepak

Please re-enable the test which have been suppressed.
https://github.com/prestodb/presto/pull/19548/commits/a297f978b5188d9c4529ce73dccb846bdfeb2191

```
Caused by: VeloxRuntimeError:  not a scalar type! kind: HUGEINT
	at Unknown.# 0  _ZN8facebook5velox7process10StackTraceC1Ei(Unknown Source)
	at Unknown.# 1  

Error:  Failures: 
Error:    TestPrestoNativeGeneralQueriesJSON>AbstractTestNativeGeneralQueries.testDecimalArithmetic:404->AbstractTestQueryFramework.assertQuery:144->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT n + m from (values (DECIMAL'999999999999999.999' , DECIMAL'1'),(DECIMAL'-123456789012345.123456', DECIMAL'-9999999999999999'),(DECIMAL'1.23', DECIMAL'-0.0005'),(DECIMAL'1.33333333333333333333', DECIMAL'-0.0005'),(NULL, NULL), (decimal'1.23', NULL)) t(n, m)
Error:    TestPrestoNativeGeneralQueriesJSON>AbstractTestNativeGeneralQueries.testDecimalLiterals:381->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT * from (values decimal'123.12', decimal'-0.0004', NULL,CAST('922337203685477580' as DECIMAL(18,0)), decimal'1123', NULL)
Error:    TestPrestoNativeGeneralQueriesJSON>AbstractTestNativeGeneralQueries.testDecimalLogicalFunctions:479->AbstractTestQueryFramework.assertQuery:144->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT c0 from (values DECIMAL'-1.54455555555555555555',DECIMAL'3.141592653589793238', NULL, DECIMAL'2.718281828459045') t(c0) where c0 between DECIMAL'-2.0' and DECIMAL'3.0'
Error:    TestPrestoNativeGeneralQueriesThrift>AbstractTestNativeGeneralQueries.testDecimalArithmetic:404->AbstractTestQueryFramework.assertQuery:144->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT n + m from (values (DECIMAL'999999999999999.999' , DECIMAL'1'),(DECIMAL'-123456789012345.123456', DECIMAL'-9999999999999999'),(DECIMAL'1.23', DECIMAL'-0.0005'),(DECIMAL'1.33333333333333333333', DECIMAL'-0.0005'),(NULL, NULL), (decimal'1.23', NULL)) t(n, m)
Error:    TestPrestoNativeGeneralQueriesThrift>AbstractTestNativeGeneralQueries.testDecimalLiterals:381->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT * from (values decimal'123.12', decimal'-0.0004', NULL,CAST('922337203685477580' as DECIMAL(18,0)), decimal'1123', NULL)
Error:    TestPrestoNativeGeneralQueriesThrift>AbstractTestNativeGeneralQueries.testDecimalLogicalFunctions:479->AbstractTestQueryFramework.assertQuery:144->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT c0 from (values DECIMAL'-1.54455555555555555555',DECIMAL'3.141592653589793238', NULL, DECIMAL'2.718281828459045') t(c0) where c0 between DECIMAL'-2.0' and DECIMAL'3.0'
Error:    TestPrestoSparkNativeGeneralQueries>AbstractTestNativeGeneralQueries.testDecimalArithmetic:404->assertQuery:40->AbstractTestQueryFramework.assertQuery:144->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT n + m from (values (DECIMAL'999999999999999.999' , DECIMAL'1'),(DECIMAL'-123456789012345.123456', DECIMAL'-9999999999999999'),(DECIMAL'1.23', DECIMAL'-0.0005'),(DECIMAL'1.33333333333333333333', DECIMAL'-0.0005'),(NULL, NULL), (decimal'1.23', NULL)) t(n, m)
Error:    TestPrestoSparkNativeGeneralQueries>AbstractTestNativeGeneralQueries.testDecimalLiterals:381->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT * from (values decimal'123.12', decimal'-0.0004', NULL,CAST('922337203685477580' as DECIMAL(18,0)), decimal'1123', NULL)
Error:    TestPrestoSparkNativeGeneralQueries>AbstractTestNativeGeneralQueries.testDecimalLogicalFunctions:479->assertQuery:40->AbstractTestQueryFramework.assertQuery:144->AbstractTestQueryFramework.assertQuery:149 Execution of 'actual' query failed: SELECT c0 from (values DECIMAL'-1.54455555555555555555',DECIMAL'3.141592653589793238', NULL, DECIMAL'2.718281828459045') t(c0) where c0 between DECIMAL'-2.0' and DECIMAL'3.0'
[INFO] 
Error:  Tests run: 256, Failures: 9, Errors: 0, Skipped: 0
```

**Comment by mbasmanova** [comment]
@majetideepak Deepak, would you help take a look? This is blocking all PRs.

CC: @miaoever

**Comment by karteekmurthys** [comment]
https://github.com/prestodb/presto/commit/bd3b99a1f3c94a5b1210bc71cbefe36981647d85
velox version was advanced 40 minutes ago and @majetideepak  change was included with that. Is it still failing after the rebase?

**Comment by majetideepak** [comment]
@karteekmurthys here is a link to a failure. Can you take a look?
https://github.com/prestodb/presto/actions/runs/4887097385/jobs/8723278190?pr=19563

**Comment by karteekmurthys** [comment]
That job was triggered by PR: https://github.com/prestodb/presto/pull/19563 created 8 hrs ago. The velox version was advanced 6 hrs ago. This branch needs to be rebased: https://github.com/rschlussel/presto/commits/aqe-stats-2.

Also, I rebased on local machine and ran the tests. They all passed.
<img width="613" alt="Screen Shot 2023-05-04 at 10 24 55 PM" src="https://user-images.githubusercontent.com/6971561/236383041-e912b81c-c7d5-4dbc-aad5-2b15985c1349.png">

**Comment by karteekmurthys** [comment]
I am able to reproduce the failure locally:

```
2023-05-05T00:51:19.290-0500	ERROR	remote-task-callback-2	com.facebook.presto.execution.StageExecutionStateMachine	Stage execution 20230505_055118_00000_j9kcp.0.0 failed
VeloxRuntimeError:  not a scalar type! kind: HUGEINT
	at Unknown.# 0  (Unknown Source)
	at Unknown.# 1  (Unknown Source)
	at Unknown.# 2  (Unknown Source)
	at Unknown.# 3  (Unknown Source)
	at Unknown.# 4  (Unknown Source)
```
Will keep this issue posted with any findings.

**Comment by karteekmurthys** [comment]
Root cause is HUGEINT is not supported in the VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH.

```
E20230504 22:59:27.301389 1033861 Exceptions.h:68] Line: /Users/karteek/workspace/oss_presto/presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:787, Function:operator(), Expression:  not a scalar type! kind: HUGEINT, Source: RUNTIME, ErrorCode: INVALID_STATE
I20230504 22:59:27.308184 1033861 TaskManager.cpp:401] Deleting task 20230505_055926_00000_jknbg.0.0.0
```

Error occurs in this method:
```
void setCellFromVariant(
    const RowVectorPtr& data,
    vector_size_t row,
    vector_size_t column,
    const velox::variant& value) {
  auto columnVector = data->childAt(column);
  if (value.isNull()) {
    columnVector->setNull(row, true);
    return;
  }
  VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
      setCellFromVariantByKind,
      columnVector->typeKind(),
      columnVector,
      row,
      value);
}
```

**Comment by majetideepak** [comment]
@karteekmurthys can you specialize `setCellFromVariant` for Hugeint and get CI to pads?
I am working on the Velox side to get the dispatch template support Hugeint.

**Comment by karteekmurthys** [comment]
@majetideepak already on it.

**Comment by karteekmurthys** [comment]
Second issue with tests is the decimal overflow cases expects substring "Decimal overflow" in the error message similar to java. Currently, in velox we have "Value '{}' is not in the range of Decimal Type".

**Comment by karteekmurthys** [comment]
I was able to fix the above 2 issues and run the tests locally. Opened one PR in presto https://github.com/prestodb/presto/pull/19574 and another in velox https://github.com/facebookincubator/velox/pull/4846.

**Comment by majetideepak** [comment]
I have another PR here to quickly unblock the CI failure.
https://github.com/prestodb/presto/pull/19577/files

**Comment by majetideepak** [comment]
Resolved with #19577
